src/main/java/com/google/sps/data/Event.java

```diff
@@ -48,119 +45,54 @@
+   * @param startTime start of activity (YYYY-MM-DD'T'HH:MM:SS)
    * @param travelTime time spent traveling to next location (minutes). 
    *        Null if last location of the day.
    * @param timeAtLocation time spent at POI (minutes)
    */
-  public Event(String name, String address, String date, int startTime, 
+  public Event(String name, String address, LocalDateTime startTime, 
               int travelTime, int timeAtLocation) {
     this.name = name;
     this.address = address;
-    this.date = date;
     this.startTime = startTime;
-    this.travelTime = travelTime;
-    this.endTime = calculateEndTime(timeAtLocation);
-  
-    createCalendarTimes();
+    this.endTime = startTime.plusMinutes(Long.valueOf(timeAtLocation));
+    this.travelTime = Long.valueOf(travelTime);
+    checkTravelTime(this.travelTime);
   }
- 
+
   /**
-   * Constructor that assume one hour default spent at location
+   * Constructor that takes in time spent at location
    * 
    * @param name name of the location (NOT address)
    * @param address exact address of the POI
-   * @param date date from user input travel day. Cannot be null.
-   * @param startTime start of activity (HHMM format)
-   * @param travelTime time spent traveling to next location (minutes)
+   * @param startTime start of activity (YYYY-MM-DD'T'HH:MM:SS)
+   * @param travelTime time spent traveling to next location (minutes). 
    *        Null if last location of the day.
    */
-  public Event(String name, String address, String date, int startTime, int travelTime) {
-    this(name, address, date, startTime, travelTime, HOUR);
-  }
-
-  // function that sets start and end Time with correct string format
-  private void createCalendarTimes() {
-    try {
-      this.strStartTime = createStrTime(this.date, this.startTime);
-      this.strEndTime = createStrTime(this.date, this.endTime);
-    } catch (Exception e) {
-      e.printStackTrace();
-    }
+  public Event(String name, String address, LocalDateTime startTime, 
+              int travelTime) {
+    this(name, address, startTime, travelTime, HOUR);
   }
 
   /**
-   * function that builds string w correct format for calendar-script.js
+   * checks that user inputs travelTime is valid
    * 
-   * @param date date from user input travel day. Cannot be null.
-   * @param time time to to start or end event. Must be within bounds
+   * @param time amount of time spent traveling (minutes)
    */
-  private static String createStrTime(String date, int time) throws Exception {
-    if (date == null) {
-      throw new NullPointerException("Date cannot be null");
-    }
-
+  private static void checkTravelTime(long time) {
     if (time < MIN_POSSIBLE_TIME) {
       throw new IllegalArgumentException("Time cannot be less than " + MIN_POSSIBLE_TIME);
     }
 
     if (time >= MINUTES_IN_A_DAY) {
       throw new IllegalArgumentException("Time cannot be more than " + MINUTES_IN_A_DAY);
```

Comment:
I see you didn't changed than, but should the error be "Time cannot be more than **or equal to** " + MINUTES_IN_A_DAY?